internal/pkg/term/signal/signal.go

```diff
@@ -0,0 +1,36 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package signal provides a way to capture and handle operating system signals.
+package signal
+
+import (
+	"os"
+	"os/signal"
+)
+
+// Signal represents a signal handler for capturing and handling operating system signals.
+type Signal struct {
+	signalCh chan os.Signal
+	sigs     []os.Signal
+}
+
+// NewSignal creates a new Signal object with the specified signals.
+func NewSignal(signals ...os.Signal) *Signal {
+	return &Signal{
+		signalCh: make(chan os.Signal),
+		sigs:     signals,
+	}
+}
+
+// NotifySignals starts capturing the specified signals and returns a channel for receiving them.
+func (s *Signal) NotifySignals() <-chan os.Signal {
+	signal.Notify(s.signalCh, s.sigs...)
+	return s.signalCh
+}
+
+// StopCatchSignals stops capturing signals and closes the signalChannel.
+func (s *Signal) StopCatchSignals() {
```

Comment:
How do you plan to use these methods 🤔 My worry is these methods won't be useful enough as they are pretty simple and we might need to make these wrap up more logic when handling signals.

Reply:
Agee with you. I will add more logic to this PR that will handle the whole `ctrl-c` for all the workloads. But the only concern may be PR will be big.